certbot/util.py

```diff
@@ -277,77 +283,58 @@ def get_filtered_names(all_names):
             logger.debug('Not suggesting name "%s"', name, exc_info=True)
     return filtered_names
 
-
-def get_os_info(filepath="/etc/os-release"):
+def get_os_info():
     """
     Get OS name and version
 
-    :param str filepath: File path of os-release file
     :returns: (os_name, os_version)
     :rtype: `tuple` of `str`
     """
 
-    if os.path.isfile(filepath):
-        # Systemd os-release parsing might be viable
-        os_name, os_version = get_systemd_os_info(filepath=filepath)
-        if os_name:
-            return os_name, os_version
-
-    # Fallback to platform module
-    return get_python_os_info()
-
+    return get_python_os_info(pretty=False)
 
-def get_os_info_ua(filepath="/etc/os-release"):
+def get_os_info_ua():
     """
     Get OS name and version string for User Agent
 
-    :param str filepath: File path of os-release file
     :returns: os_ua
     :rtype: `str`
     """
+    if USE_DISTRO:
+        os_info = distro.name(pretty=True)
 
-    if os.path.isfile(filepath):
-        os_ua = get_var_from_file("PRETTY_NAME", filepath=filepath)
-        if not os_ua:
-            os_ua = get_var_from_file("NAME", filepath=filepath)
-        if os_ua:
-            return os_ua
-
-    # Fallback
-    return " ".join(get_python_os_info())
+    if not USE_DISTRO or not os_info:
+        return " ".join(get_python_os_info(pretty=True))
+    return os_info
 
-
-def get_systemd_os_info(filepath="/etc/os-release"):
+def get_systemd_os_info():
     """
     Parse systemd /etc/os-release for distribution information
 
-    :param str filepath: File path of os-release file
     :returns: (os_name, os_version)
     :rtype: `tuple` of `str`
     """
 
-    os_name = get_var_from_file("ID", filepath=filepath)
-    os_version = get_var_from_file("VERSION_ID", filepath=filepath)
+    warnings.warn(
+        "The get_sytemd_os_like() function is deprecated and will be removed in "
+        "a future release.", DeprecationWarning, stacklevel=2)
+    return get_os_info()
 
-    return (os_name, os_version)
 
-
-def get_systemd_os_like(filepath="/etc/os-release"):
+def get_systemd_os_like(filepath="/etc/os-release"):  # pylint: disable=unused-argument
```

Comment:
I think we should just delete this `filepath` argument since it's not used anymore.

Reply:
You are right, as the parameter isn't doing anything, I ripped it off. Better to fail early there if it's used in a third party plugin.